mongo/datadog_checks/mongo/mongo.py

```diff
@@ -457,9 +457,22 @@ def _report_replica_set_state(self, state, clean_server_name, replset_name):
         * Create an event on state change.
         """
         last_state = self._last_state_by_server.get(clean_server_name, -1)
+
+        self.log.debug(
+            'report_replica_set_state state=%r last_state=%r clean_server_name=%r replset_name=%r',
+            state,
+            last_state,
+            clean_server_name,
+            replset_name,
+        )
+
         self._last_state_by_server[clean_server_name] = state
-        if last_state != state and last_state != -1:
-            return self.create_event(last_state, state, clean_server_name, replset_name)
+
+        state_changed = last_state != state and last_state != -1
+        if not state_changed:
```

Comment:
Why not
```
if state_changed:
    self.create_event(last_state, state, clean_server_name, replset_name)
```
A return there almost at the end feels a bit silly

Reply:
I tend to take the [guard clauses](https://dev.to/wagslane/guard-clauses-how-to-clean-up-conditionals-2fdm) approach to conditionals, this is an example of it. 😃